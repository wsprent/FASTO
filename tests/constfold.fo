fun int main () =
  let x0 = write(2+5) in
  let x1 = write(x0 - 2) in
  let x2 = write(x1 * 4) in
  let x3 = x2 in
  let x4 = write(x3 == x2) in
  write(x2 / x1)