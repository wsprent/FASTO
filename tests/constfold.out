7 5 20 True4